File: InstitutoTriboDeDavi.FLUTTER/flutter_tribo_de_davi_api/lib/entities/models/polo.dart
class Polo {
  final int id;
  final String nome;
  final String informacoes;
  final String endereco;
  final String? bairro;
  final String? cidade;

  const Polo(
      {required this.id,
      required this.nome,
      required this.informacoes,
      required this.endereco,
      this.bairro,
      this.cidade});

  const Polo.empty(
      {this.id = 0,
      this.nome = '',
      this.informacoes = '',
      this.endereco = '',
      this.bairro = '',
      this.cidade = ''});

  factory Polo.fromJson(Map<String, dynamic> json) => Polo(
      id: json['id'],
      nome: json['nome'],
      informacoes: json['informacoes'],
      endereco: json['endereco'],
      bairro: json['bairro'],
      cidade: json['cidade']);

  Map<String, dynamic> toJson() => {
        "id": id,
        "nome": nome,
        "informacoes": informacoes,
        "endereco": endereco,
        "bairro": bairro,
        "cidade": cidade
      };
}

File: InstitutoTriboDeDavi.Domain/Entities/Polo.cs
using InstitutoTriboDeDavi.Domain.Common;
using InstitutoTriboDeDavi.Domain.Exceptions;
using InstitutoTriboDeDavi.Domain.Validators;

namespace InstitutoTriboDeDavi.Domain.Entities
{
    public class Polo : Base
    {
        public string Nome { get; set; }
        public string Informacoes { get; set; }
        public string Endereco { get; set; }
        public string Bairro { get; set; }
        public string Cidade { get; set; }

        // Limite de alunos ativos no ano. 0 = sem limite. Ao atingir, novas
        // inscrições para este polo são bloqueadas até liberar vaga (inativar
        // matrícula ou aumentar o limite).
        public int LimiteAlunos { get; set; }

        // O polo tem turma de adultos? Quando false, a inscrição de adultos é
        // bloqueada neste polo (a ficha infantil segue permitida). Default true
        // para não mudar o comportamento dos polos já existentes.
        public bool AceitaAdultos { get; set; } = true;

        // Horários de treino por turma (um item por dia da semana).
        public List<HorarioTurma> Horarios { get; set; } = new();

        public override bool Validate()
        {
            var validator = new PoloValidator();
            var validation = validator.Validate(this);

            if (!validation.IsValid)
            {
                foreach (var error in validation.Errors)
                {
                    _errors.Add(error.ErrorMessage);
                }

                throw new DomainException("Alguns campos estão inválidos!", _errors);
            }

            return true;
        }
    }
}
